modin/engines/ray/generic/io.py

```diff
@@ -303,13 +307,19 @@ def _read_csv_from_file_pandas_on_ray(cls, filepath, kwargs={}):
             # step has to happen without removing the `index_col` otherwise it will not
             # be assigned correctly
             kwargs["index_col"] = None
+            f = file_open(filepath, "rb", kwargs)
+            kwargs_uncompressed = kwargs.copy()
+            kwargs_uncompressed["compression"] = None
             names = pandas.read_csv(
-                file_open(filepath, "rb"), **dict(kwargs, nrows=0, skipfooter=0)
+                f, **dict(kwargs_uncompressed, nrows=0, skipfooter=0)
```

Comment:
Can you add some inline comments about why the compression must be `None` for the single line case?